scenario_player/runner.py

```diff
@@ -2,42 +2,172 @@
 
 import gevent
 import structlog
 from eth_typing import ChecksumAddress
-from eth_utils import encode_hex, is_checksum_address, to_canonical_address, to_checksum_address
-from raiden_contracts.contract_manager import ContractManager, contracts_precompiled_path
-from requests import HTTPError, RequestException, Session
+from eth_utils import (
+    encode_hex,
+    is_checksum_address,
+    to_bytes,
+    to_canonical_address,
+    to_checksum_address,
+)
+from raiden_contracts.constants import CONTRACT_USER_DEPOSIT
+from raiden_contracts.contract_manager import (
+    ContractManager,
+    contracts_precompiled_path,
+    get_contracts_deployment_info,
+)
+from requests import HTTPError, Session
 from web3 import HTTPProvider, Web3
-from web3.contract import Contract
 
 from raiden.accounts import Account
-from raiden.constants import GAS_LIMIT_FOR_TOKEN_CONTRACT_CALL
-from raiden.network.rpc.client import JSONRPCClient
-from raiden.utils.typing import TransactionHash
+from raiden.constants import UINT256_MAX
+from raiden.network.proxies.proxy_manager import ProxyManager
+from raiden.network.proxies.token import Token
+from raiden.network.proxies.token_network_registry import TokenNetworkRegistry
+from raiden.network.proxies.user_deposit import UserDeposit
+from raiden.network.rpc.client import EthTransfer, JSONRPCClient
+from raiden.settings import RAIDEN_CONTRACT_VERSION
+from raiden.utils.typing import TransactionHash, UserDepositAddress
 from scenario_player.constants import (
     API_URL_TOKEN_NETWORK_ADDRESS,
-    API_URL_TOKENS,
     NODE_ACCOUNT_BALANCE_FUND,
     NODE_ACCOUNT_BALANCE_MIN,
     OWN_ACCOUNT_BALANCE_MIN,
     RUN_NUMBER_FILENAME,
 )
 from scenario_player.definition import ScenarioDefinition
-from scenario_player.exceptions import ScenarioError, TokenRegistrationError
+from scenario_player.exceptions import ScenarioError
 from scenario_player.exceptions.legacy import TokenNetworkDiscoveryTimeout
-from scenario_player.services.utils.interface import ServiceInterface
-from scenario_player.utils import TimeOutHTTPAdapter, get_udc_and_token, wait_for_txs
-from scenario_player.utils.token import Token, UserDepositContract
+from scenario_player.node_support import NodeController, RaidenReleaseKeeper
+from scenario_player.utils import TimeOutHTTPAdapter, wait_for_txs
+from scenario_player.utils.configuration.settings import SettingsConfig, UDCSettingsConfig
+from scenario_player.utils.token import (
+    load_token_configuration_from_file,
+    token_maybe_mint,
+    token_registry_maybe_register_token,
+    userdeposit_maybe_deposit,
+    userdeposit_maybe_increase_allowance,
+)
 
 if TYPE_CHECKING:
     from scenario_player.tasks.base import Task, TaskState
 
 log = structlog.get_logger(__name__)
 
 
+def get_udc_and_corresponding_token(
+    udc_settings: UDCSettingsConfig, settings: SettingsConfig, proxy_manager: ProxyManager
+) -> Tuple[UserDeposit, Token]:
```

Comment:
This corresponds to the old [`get_udc_and_token`](https://github.com/raiden-network/scenario-player/blob/f208ed6cb4f713c1efc921b10458365bcf918b5d/scenario_player/utils/legacy.py#L301-L326), however now it returns the proxies from Raiden instead of yet another proxy.

The type signature of the function was slightly simplified by always requiring the `assert udc_settings.enable`, which means the return type does not need `Optional` anymore.